File: debot/src/main/java/com/tomoima/debot/adapter/DebotMenuListAdapter.java
package com.tomoima.debot.adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.tomoima.debot.R;
import com.tomoima.debot.strategy.DebotStrategy;

import java.util.List;


public class DebotMenuListAdapter extends ArrayAdapter<DebotStrategy> {
    private LayoutInflater inflater;

    public DebotMenuListAdapter(Context context, List<DebotStrategy> objects) {
        super(context, -1, objects);
        inflater = LayoutInflater.from(context);
    }

    @NonNull
    @Override
    public View getView(int position, View view, ViewGroup parent) {
        ViewHolder vh;
        if (view == null) {
            view = inflater.inflate(R.layout.row_debug_menu, parent, false);
            vh = new ViewHolder(view);
            view.setTag(vh);
        } else {
            vh = (ViewHolder) view.getTag();
        }
        vh.menuTitle.setText(getItem(position).getStrategyMenuName());
        return view;
    }


    private static class ViewHolder {
        TextView menuTitle;

        ViewHolder(View view) {
            menuTitle = (TextView) view.findViewById(R.id.debug_menu_title);
        }
    }
}

File: debot/src/main/java/com/tomoima/debot/adapter/DebotMenuListAdapter.kt
package com.tomoima.debot.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView

import com.tomoima.debot.R
import com.tomoima.debot.strategy.DebotStrategy


class DebotMenuListAdapter(context: Context, objects: List<DebotStrategy>) : ArrayAdapter<DebotStrategy>(context, -1, objects) {
    private val inflater: LayoutInflater = LayoutInflater.from(context)

    override fun getView(position: Int, view: View?, parent: ViewGroup): View {
        val v: View
        val vh: ViewHolder

        if (view == null) {
            v = inflater.inflate(R.layout.row_debug_menu, parent, false)
            vh = ViewHolder(v)
            v?.tag = vh
        } else {
            v = view
            vh = view.tag as ViewHolder
        }
        vh.menuTitle.text = getItem(position)?.strategyMenuName
        return v
    }


    private class ViewHolder internal constructor(view: View) {
        internal var menuTitle = view.findViewById(R.id.debug_menu_title) as TextView

    }
}
